lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -100,6 +100,35 @@ def add_order(post, money, options, capture: false)
         post['order']['references']['invoiceData'] = {
           'invoiceNumber' => options[:invoice]
         }
+        add_airline_data(post, options) if options[:airline_data]
+      end
+
+      def add_airline_data(post, options)
+        airline_data = {}
+
+        flight_date = options[:airline_data][:flight_date]
+        passenger_name = options[:airline_data][:passenger_name]
+        code = options[:airline_data][:code]
+        name = options[:airline_data][:name]
+
+        airline_data['flightDate'] = flight_date if flight_date
+        airline_data['passengerName'] = passenger_name if passenger_name
+        airline_data['code'] = code if code
+        airline_data['name'] = name if name
+
+        flight_legs = []
+        options[:airline_data][:flight_legs].each do |fl|
+          leg = {}
+          leg['arrivalAirport'] = fl[:arrival_airport] if fl[:arrival_airport]
+          leg['originAirport'] = fl[:origin_airport] if fl[:origin_airport]
+          leg['date'] = fl[:date] if fl[:date]
+          leg['number'] = fl[:number] if fl[:number]
+          leg['carrierCode'] = fl[:carrier_code] if fl[:carrier_code]
+          leg['airlineClass'] = fl[:carrier_code] if fl[:airline_class]
+          flight_legs << leg
+        end
+        airline_data['flightLegs'] = flight_legs
+        post['order']['additionalInput']['airlineData'] = airline_data
```

Comment:
Because all this method is really doing is converting a Ruby hash into JSON, I think that the whole method can be replaced with the following:

```
def add_airline_data(post, options)
  airline_data = options[:airline_data].deep_stringify_keys.deep_transform_keys { |key| key.camelize(:lower) }
  post['order']['additionalInput']['airlineData'] = airline_data
end
```

I've tested this with the unit and remote tests, and they all still pass. What do you think?

Reply:
I thought about trying something like this but feel anxious about allowing users to send arbitrary data to an endpoint via our servers. 😬

Reply:
Which I realize we're kind of doing anyway, but it still makes me 😬

Reply:
That's a good point. Do you know if there's a way to apply strong parameters to something like this to apply whitelisting?

Reply:
🤔 I think of `strong params` as being a Rails (or maybe general DB-backed app) thing that controllers handle when creating DB records, which isn't what we're doing here. So on the one hand, I don't know of a way. On the other hand, hopefully this _is_ being validated on the DB side, so maybe my concerns are overblown. (But on the third hand, I'd just as soon not find out?) 

In the future case, I continue to muse intermittently on the possibility of some generalizable solution for GSFs, which means this is a problem we may still want to solve eventually.

Reply:
Yeah, it's something that I've been mulling over for a while as I keep thinking that there has to be a better way... Thanks for sharing your thoughts! :grin: